ms-engine/src/main/java/net/trajano/ms/engine/internal/resteasy/SemaphoredHeaders.java

```diff
@@ -0,0 +1,63 @@
+package net.trajano.ms.engine.internal.resteasy;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Semaphore;
+
+import org.jboss.resteasy.core.Headers;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This will prevent "gets" until a the lock is released.
+ *
+ * @author Archimedes Trajano
+ * @param <V>
+ */
+public class SemaphoredHeaders<V> extends Headers<V> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override the "equals" method in this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2160)